pkg/registry/apps/statefulset/registry.go

```diff
@@ -73,15 +73,15 @@ func (s *storage) GetStatefulSet(ctx genericapirequest.Context, statefulSetID st
 }
 
 func (s *storage) CreateStatefulSet(ctx genericapirequest.Context, statefulSet *apps.StatefulSet) (*apps.StatefulSet, error) {
-	obj, err := s.Create(ctx, statefulSet, false)
+	obj, err := s.Create(ctx, statefulSet, rest.ValidateAllObjectFunc, false)
```

Comment:
So I've been reading through this code.  I think it is a mistake to make this intermediate registry for use by the scale client.  It makes what is happening far less obvious.  Your choice here is safe, but only because of how this is used.

Please open an issue.

Reply:
It's boilerplate code essentially doing a type cast, not more. I agree this complicates more than it helps.

Reply:
https://github.com/kubernetes/kubernetes/issues/54693

Reply:
Following the old behaviour, I wired the validation functions through now from the subresource registries via the intermediate registries to the storage.